MIN aggregation function panics on query against CSUP with named type

## tl;dr

Given test data `data.jsup`:

```
{EventDate:2013-07-15T00:00:00Z(=arrow_date32)}
{EventDate:2013-07-16T00:00:00Z(=arrow_date32)}
```

The following query in vector runtime causes a panic.

```
$ super -f csup -o data.csup data.jsup && SUPER_VAM=1 super -c 'SELECT MIN(EventDate) AS min_EventDate FROM "data.csup"'
panic: (*vector.Named) 0xc000c06060

goroutine 65 [running]:
github.com/brimdata/super/runtime/vam/expr/agg.minFlat[...](0xc000c0c020?, {0xe9a4d50?, 0xc000c06060?}, {0x0?, 0xddf95f6?, 0xe9a4b58?})
	/Users/phil/work/super/runtime/vam/expr/agg/mathfunc.go:117 +0x129
github.com/brimdata/super/runtime/vam/expr/agg.min[...](0x0?, {0xe9a4d50?, 0xc000c06060?})
	/Users/phil/work/super/runtime/vam/expr/agg/mathfunc.go:104 +0xf4
github.com/brimdata/super/runtime/vam/expr/agg.(*reduceInt64).consume(0xc000c000a0, {0xe9a4d50?, 0xc000c06060?})
	/Users/phil/work/super/runtime/vam/expr/agg/math.go:164 +0x2c
github.com/brimdata/super/runtime/vam/expr/agg.(*mathReducer).Consume(0xc000c00080, {0xe9a4d50, 0xc000c06060})
	/Users/phil/work/super/runtime/vam/expr/agg/math.go:78 +0x316
github.com/brimdata/super/runtime/vam/op/summarize.(*superTable).update(0xc000c02050, {0xc000c1c030, 0x0, 0x1}, {0xc000c1c030, 0x1, 0xc000c1df80?})
	/Users/phil/work/super/runtime/vam/op/summarize/agg.go:70 +0x659
github.com/brimdata/super/runtime/vam/op/summarize.(*Summarize).consume(0xc0009d6000, {0xc000c1c030, 0x0, 0x1}, {0xc000c1c030, 0x1, 0x1})
	/Users/phil/work/super/runtime/vam/op/summarize/summarize.go:102 +0x4cc
github.com/brimdata/super/runtime/vam/op/summarize.(*Summarize).Pull.func1({0xc000c1c030, 0x1, 0xc000481d80?})
	/Users/phil/work/super/runtime/vam/op/summarize/summarize.go:83 +0x78
github.com/brimdata/super/vector.Apply(0x0?, 0xc000c1c030?, {0xc000c1c030?, 0xc000c1c020?, 0xe79ec40?})
	/Users/phil/work/super/vector/apply.go:17 +0xa3
github.com/brimdata/super/runtime/vam/op/summarize.(*Summarize).Pull(0xc0009d6000, 0x0?)
	/Users/phil/work/super/runtime/vam/op/summarize/summarize.go:82 +0x13b
github.com/brimdata/super/runtime/vam/op.(*combineParent).run(0xc0009ac9c0)
	/Users/phil/work/super/runtime/vam/op/combine.go:110 +0x3f
created by github.com/brimdata/super/runtime/vam/op.(*Combine).Pull.func1 in goroutine 1
	/Users/phil/work/super/runtime/vam/op/combine.go:42 +0x35
```

## Details

Repro is with `super` commit a69b860. The query and data are a simplification of the [ClickBench q6 query](https://github.com/ClickHouse/ClickBench/blob/acdd156c11043d45e9e54f791747bf51e9564d06/duckdb/queries.sql#L7).

When run in sequential runtime, there's no panic.

```
$ super -version
Version: v1.18.0-270-ga69b860e

$ super -f csup -o data.csup data.jsup && super -c 'SELECT MIN(EventDate) AS min_EventDate FROM "data.csup"'
{min_EventDate:2013-07-15T00:00:00Z}
```

It also doesn't panic in vector runtime if I do a cast to `time` to strip out the named type.

```
$ super -f csup -o data.csup -c 'EventDate:=time(EventDate)' data.jsup && SUPER_VAM=1 super -c 'SELECT MIN(EventDate) AS min_EventDate FROM "data.csup"'
{min_EventDate:2013-07-15T00:00:00Z}
```

It only panics if I'm querying CSUP in vector runtime. No panic on Parquet.

```
$ super -f parquet -o data.parquet data.jsup && SUPER_VAM=1 super -c 'SELECT MIN(EventDate) AS min_EventDate FROM "data.parquet"' 
{min_EventDate:2013-07-15T00:00:00Z}
```

--- a/runtime/vam/expr/agg/avg.go
+++ b/runtime/vam/expr/agg/avg.go
@@ -14,6 +14,7 @@ type avg struct {
 var _ Func = (*avg)(nil)
 
 func (a *avg) Consume(vec vector.Any) {
+	vec = vector.Under(vec)
 	if !super.IsNumber(vec.Type().ID()) {
 		return
 	}

--- a/runtime/vam/expr/agg/math.go
+++ b/runtime/vam/expr/agg/math.go
@@ -38,6 +38,7 @@ func (m *mathReducer) Result(*super.Context) super.Value {
 }
 
 func (m *mathReducer) Consume(vec vector.Any) {
+	vec = vector.Under(vec)
 	typ := vec.Type()
 	var id int
 	if m.math != nil {

--- a/runtime/ztests/op/summarize/named-values.yaml
+++ b/runtime/ztests/op/summarize/named-values.yaml
@@ -1,0 +1,24 @@
+zed: count(), sum(this), min(this), max(this), avg(this), collect(this), any(this), dcount(this)
+
+vector: true
+
+input: | 
+  1(=named)
+  2(=named)
+
+output-flags: -pretty=4
+
+output: |
+  {
+      count: 2 (uint64),
+      sum: 3,
+      min: 1,
+      max: 2,
+      avg: 1.5,
+      collect: [
+          1,
+          2
+      ],
+      any: 1 (=named),
+      dcount: 2 (uint64)
+  }

